examples/petsc/area.c

```diff
@@ -136,8 +188,20 @@ int main(int argc, char **argv) {
   CHKERRQ(ierr);
   ierr = PetscOptionsEnd(); CHKERRQ(ierr);
 
+  // Setup function pointer for geometric factors
+  int (*geomfp)(void *ctx, const CeedInt Q, const CeedScalar *const *in,
+                CeedScalar *const *out);
+  ierr = PetscFunctionListFind(geomfactorlist, problemtype,
+                               (void(* *)(void))&geomfp); CHKERRQ(ierr);
+  const char *str;
+  if (geomfp == SetupMassGeoCube)
+    str = SetupMassGeoCube_loc;
+  else if (geomfp == SetupMassGeoSphere)
+    str = SetupMassGeoSphere_loc;
+  else
+    return CeedError(ceed, 1, "Function not found in the list");
```

Comment:
I missed this in my review, but the Ceed is not available here for error handling.

It looks like the typical way to handle this on the PETSc side would be https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Sys/SETERRQ.html#SETERRQ

Reply:
Thanks Jeremy! I can take care of this in my `bps-on-a-sphere` branch